### PR TITLE
Fix hessian codec bugs

### PR DESCRIPTION
**Pull request by zonghaishang**
**What this PR does**:

**Which issue(s) this PR fixes**:

Fixes https://github.com/apache/dubbo-go-hessian2/issues/164
Fixes https://github.com/apache/dubbo-go-hessian2/issues/168

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: object.go [hunk 1]
```diff
@@ -541,11 +541,17 @@ func (d *Decoder) decEnum(javaName string, flag int32) (JavaEnum, error) {
...
+	len := len(cls.fieldNameList)
+	if len < 1 {
 		return nil
 	}
-	_, err := d.DecodeValue()
-	return err
+
+	for i := 0; i < len; i++ {
+		// skip class fields.
+		d.DecodeValue()
```
Comment (wongoo) [inline object.go]: failfast to return error when not nil

File: hessian_test.go [hunk 2]
```diff
@@ -171,3 +194,30 @@ func TestRequest(t *testing.T) {
...
+func TestHessianCodec_ReadAttachments(t *testing.T) {
+	body := &Response{
+		RspObj:      &CaseB{A: "A", B: CaseA{A: "a", B: 1, C: Case{A: "c", B: 2}}},
+		Exception:   nil,
+		Attachments: map[string]string{DUBBO_VERSION_KEY: "2.6.4"},
+	}
+	resp, err := doTestHessianEncodeHeader(t, PackageResponse, Response_OK, body)
+	assert.NoError(t, err)
+
+	pojoRegistry = POJORegistry{}
```
Comment (wongoo) [inline hessian_test.go]: why clean the pojo registry？

Comment (fangyincheng) [reply hessian_test.go]: for `ReadAttachments`, the pojos should not registry.

File: object.go (hunk 1 above)
Comment (zonghaishang) [reply object.go]: ok, I will fix it.

[the remark this answers is not in the document]
Comment (zonghaishang) [reply object.go]: done

File: hessian_test.go [hunk 3]
```diff
@@ -34,8 +34,31 @@ type Case struct {
...
+type CaseB struct {
+	A string
+	B CaseA
+}
+
+func (c *CaseB) JavaClassName() string {
+	return "com.test.caseb"
+}
+
+//func (c CaseA) JavaClassName() string {
```
[the remark this answers is not in the document]
Comment (zonghaishang) [reply hessian_test.go]: done

**Comment by zonghaishang** [comment]
@wongoo Please help check TestList test case, local run is passed

https://travis-ci.org/github/apache/dubbo-go-hessian2/builds/663769046?utm_source=github_status&utm_medium=notification

error stack:
```
=== RUN   TestList
--- FAIL: TestList (0.00s)
panic: assignment to entry in nil map [recovered]
	panic: assignment to entry in nil map
goroutine 484 [running]:
testing.tRunner.func1(0xc00039c500)
	/home/travis/.gimme/versions/go1.11.1.linux.amd64/src/testing/testing.go:792 +0x6a7
panic(0x88d280, 0x95fe00)
	/home/travis/.gimme/versions/go1.11.1.linux.amd64/src/runtime/panic.go:513 +0x1b9
github.com/apache/dubbo-go-hessian2.RegisterPOJO(0x960880, 0xc10c20, 0x0)
	/home/travis/build/apache/dubbo-go-hessian2/pojo.go:149 +0x2f9
github.com/apache/dubbo-go-hessian2.RegisterPOJOs(0xc0001b9f38, 0x3, 0x3, 0xc0000a4cc0, 0x871c00, 0x95e650)
	/home/travis/build/apache/dubbo-go-hessian2/pojo.go:218 +0xf6
github.com/apache/dubbo-go-hessian2.TestList(0xc00039c500)
	/home/travis/build/apache/dubbo-go-hessian2/list_test.go:75 +0x125
testing.tRunner(0xc00039c500, 0x906fc8)
	/home/travis/.gimme/versions/go1.11.1.linux.amd64/src/testing/testing.go:827 +0x163
created by testing.(*T).Run
	/home/travis/.gimme/versions/go1.11.1.linux.amd64/src/testing/testing.go:878 +0x651
exit status 2
FAIL	github.com/apache/dubbo-go-hessian2	30.313s
```

File: hessian_test.go (hunk 2 above)
Comment (wongoo) [reply hessian_test.go]: the pojo should be recovered at the end of the unit test

Comment (wongoo) [reply hessian_test.go]: @zonghaishang  the pojo should be recovered at the end of the unit test

Comment (zonghaishang) [reply hessian_test.go]: yes, I will fix it .

File: pojo.go [hunk 4]
```diff
@@ -210,6 +210,46 @@ func RegisterPOJO(o POJO) int {
...
+	var structInfo structInfo
+	v := reflect.ValueOf(o)
+	switch v.Kind() {
+	case reflect.Struct:
+		structInfo.typ = v.Type()
+	case reflect.Ptr:
+		structInfo.typ = v.Elem().Type()
+	default:
+		structInfo.typ = reflect.TypeOf(o)
+	}
```
Comment (wongoo) [inline pojo.go]: these lines can be extracted to a new func

File: pojo.go [hunk 5]
```diff
@@ -210,6 +210,46 @@ func RegisterPOJO(o POJO) int {
 	return structInfo.index
 }
 
+// easy for test
```
Comment (wongoo) [inline pojo.go]: seems useful ， not only for test

Comment (zonghaishang) [reply pojo.go]: nice, :-)

File: pojo.go (hunk 4 above)
Comment (zonghaishang) [reply pojo.go]: I will fix it.

File: pojo.go [hunk 6]
```diff
@@ -210,6 +201,50 @@ func RegisterPOJO(o POJO) int {
...
+	}
+
+	return arr
+}
+
+func UnRegisterPOJO(o POJO) int {
+	pojoRegistry.Lock()
+	defer pojoRegistry.Unlock()
+
+	var structInfo structInfo
```
Comment (fangyincheng) [inline pojo.go]: you don't need to define a structInfo: 

`goName := obtainValueType(o).String()`

Comment (zonghaishang) [reply pojo.go]: done.

**Comment by wongoo** [comment]
LGTM

**Pull request merged**